File: 021. Palindrome Check/PalindromeCheckOnOnRecursive.java
import java.util.*;

class PalindromeCheckOnOnRecursive {
    public static boolean isPalindrome(String str) {
        return isPalindrome(str, 0);
    }

    public static boolean isPalindrome(String str, int i) {
        int j = str.length() - 1 - i;
        return i >= j ? true : str.charAt(i) == str.charAt(j) && isPalindrome(str, i + 1);
    }
}

File: 021. Palindrome Check/PalindromeCheckOnOnRecursive.kt
package com.algoexpert.program

fun PalindromeCheckOnOnRecursive(string: String): Boolean {
    return isPalindrome(string, 0)
}

fun isPalindrome(str: String, i: Int): Boolean {
    val j = str.length - 1 - i
    return if (i >= j) true else str.get(i) == str.get(j) && isPalindrome(str, i + 1)
}
